go-controller/pkg/ovn/base_network_controller_user_defined.go

```diff
@@ -539,19 +538,17 @@ func (bsnc *BaseUserDefinedNetworkController) hasIPAMClaim(pod *corev1.Pod, nadN
 		ipamClaimName, wasPersistentIPRequested = pod.Annotations[util.OvnUDNIPAMClaimName]
 	} else {
 		// secondary network the IPAM claim reference is on the network selection element
-		nadKeys := strings.Split(nadNamespacedName, "/")
-		if len(nadKeys) != 2 {
-			return false, fmt.Errorf("invalid NAD name %s", nadNamespacedName)
-		}
-		nadNamespace := nadKeys[0]
-		nadName := nadKeys[1]
-		allNetworks, err := util.GetK8sPodAllNetworkSelections(pod)
+		on, networkMap, err := util.GetPodNADToNetworkMapping(pod, bsnc)
 		if err != nil {
-			return false, err
+			return false, fmt.Errorf("failed to get network mapping for pod %s/%s on network %s: %v",
+				pod.Namespace, pod.Name, bsnc.GetNetworkName(), err)
+		}
+		if !on {
+			klog.Warningf("Pod %s/%s is not scheduled on network %s", pod.Namespace, pod.Name, bsnc.GetNetworkName())
+			return false, nil
 		}
-		for _, network := range allNetworks {
-			if network.Namespace == nadNamespace && network.Name == nadName {
-				// found network selection element, check if it has IPAM
+		for nadName, network := range networkMap {
+			if nadName == nadKey {
```

Comment:
networkMap no longer has a nadName, it's a nadKey. So you should call it something else and not compare nadName to nadKey.